File: 2015/10/10.cpp
#include <iostream>


int main (void) {

    std::string problem_input {"1321131112"}, new_seq {};
    int seq_len {1}, iters {50};

    for (int j {1}; j <= iters; j++) {
        problem_input += 'N';
        for (int i {}; i < problem_input.length()-1; i++){
            if (problem_input[i] == problem_input[i+1]) {
                seq_len++;
            }
            else {
                new_seq += std::to_string(seq_len) + problem_input[i];
                seq_len = 1;
            }
        }
        if (j == 40) {
            std::cout << new_seq.length() << std::endl;
        }
        if (j != iters) {
            problem_input = new_seq;
            new_seq = "";
        }
    }
    std::cout << new_seq.length() << std::endl;

}

// 174.9 ms ± 3.7 ms (100 runs, 10 warmup) - G++ 12.1.1: g++ -O3 -std=C++20

File: 2015/10/10.py
problem_input, new_seq = '1321131112', ''
seq_len, iters = 1, 50

for j in range(1, iters+1):
    problem_input += 'N'
    for i in range(len(problem_input)-1):
        if problem_input[i] == problem_input[i+1]:
            seq_len += 1
        else:
            new_seq += str(seq_len) + problem_input[i]
            seq_len = 1
    if j == 40:
        print(len(new_seq))
    if j != iters:
        problem_input = new_seq
        new_seq = ''
print(len(new_seq))

# 5447 ms ±  85 ms (10 runs, 2    warmup) - Python 3.10.5 (conda-forge)